OCI Connector CCP Migration

MurugeshSandari opened the pull request:
Added main CCP deployable File
Updated the PowerShell script to generate package for OCI

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below


# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

Murugesh-Sandari committed:
Added OCI CCP Connector Files
Added main CCP deployable File
Updated the PowerShell script to generate package for OCI

rahul0216 reviewed (commented):
1. Mostly looks good. Please address inline comments.
2. One query, is there any plan to update parser to include new table? The solution has existing content written using parser, those won't work with new table.

Murugesh-Sandari committed:
Update API versions and improve JSON formatting

- Updated `apiVersion` in `OCI_DataConnector_DCR.json` to `2023-03-11` and improved formatting of `streamDeclarations`.
- Changed `apiVersion` in `OCI_DataConnector_DataConnectorDefinition.json` to `2024-09-01`, reformatted instructions, and adjusted `Pass Phrase` field.
- Updated `apiVersion` in `OCI_DataConnector_PollingConfig.json` to `2024-09-01` and aligned response structure.
- Changed `apiVersion` in `OCI_DataConnector_Table.json` to `2023-09-01`.
- Adjusted `project-rename` syntax in `OCILogs.yaml` for consistency and included fuzzy matching in the `union` operation for `OCI_LogsV2_CL`.

Murugesh-Sandari committed:
Add OCI_LogsV2_CL.json schema for logging data

rahul0216 reviewed (dismissed):
Changes look good.

Ignore the hyperlink validation failure. Link has been added as example URL. It is not referring to any link.

v-prasadboke committed:
Update OCI solution to recommend CCF connector

Marked the legacy OCI Logs API data connector as deprecated and updated descriptions to recommend the new Codeless Connector Framework (CCF)-based connector. Updated solution metadata, UI, and ARM templates to reference the new connector, clarify migration guidance, and reflect the new ingestion method using the Log Ingestion API. Also updated package files and documentation to align with these changes.

v-prasadboke committed:
Update ReleaseNotes.md

v-prasadboke committed:
Update ReleaseNotes.md

v-prasadboke committed:
Update ReleaseNotes.md

v-prasadboke committed:
Update ReleaseNotes.md
